src/content/learning/using-durable-objects.md

```diff
@@ -0,0 +1,216 @@
+---
+order: 8
+---
+
+# Using Durable Objects
+
+Durable Objects provide low-latency coordination and consistent storage for the Workers platform through two features: global uniqueness and a transactional storage API.
+
+* Global Uniqueness guarantees that there will be a single Durable Object with a given id running at once.  Requests for a Durable Object id are routed by the Workers runtime to the Cloudflare point-of-presence that owns the Durable Object.  
+
+* The transactional storage API provides strongly-consistent, key-value storage to the Durable Object.  Each Object can only read and modify keys associated with that Object. Execution of a Durable Object is single-threaded, but multiple request threads may be processed out-of-order from how they arrived at the Object.
+
+## Using Durable Objects
+
+Durable Objects are named instances of a class you define.  Just like a class in object-oriented programming, the class defines the methods and data a Durable Object can access.
+
+There are four steps to creating a Durable Object:
+
+* __Writing the class__ that defines a Durable Object.
+* __Binding that class__ into a Worker.
+* __Instantiating a Durable Object__ from within a running Worker.
+* __Communicating with a Durable Object__ from a running Worker via the Fetch API.
+
+## Writing a class that defines a Durable Object
+
+To define a Durable Object, developers export an ordinary Javascript class.  Other languages will need a shim that translates their class definition to a Javascript class.
+
+The first variable passed to the class constructor contains state specific to the Durable Object, including methods for accessing storage.
+
+```js
+
+class DurableObjectExample {
+    constructor(state, environment){
+
+    }
+}
+
+```
+
+Workers communicate with a Durable Object via the fetch API.  Like any other Worker, a Durable Object listens for incoming Fetch events by registering an event handler.
+
+```js
+
+class DurableObjectExample {
+    constructor(state, environment){
+    }
+    
+    addEventListener("fetch", (event) => {
+        event.respondWith(handle(event.request));
+    });
+ 
+    async function handle(request) {
+        return new Response('Hello World');
+    }
+
+}
+
+```
+
+A Worker can pass information to a Durable Object via headers, the HTTP method, the Request body, or the Request URI.
+
+### Accessing Storage from a Durable Object
+
+Durable Objects gain access to a TODO: fill in[persistent storage API]() via the first parameter passed to the Durable Object constructor.  While access to a Durable Object is single-threaded, it's important to remember that there can still be race conditions across multiple requests.
+
+```js
+
+class DurableObjectExample {
+    constructor(state, environment){
+        this.state = state
+    }
+    
+    addEventListener("fetch", (event) => {
```

Comment:
Again, get rid of this and change the name of the function below from `handle` to `fetch`.